### PR TITLE
ddl: add telemetry support for add index acceleration

### PR DESCRIPTION
**Pull request by Benjamin2037**
### What problem does this PR solve?

Issue Number: ref: #35983

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: telemetry/data_feature_usage_test.go [hunk 1]
```diff
@@ -419,3 +420,33 @@ func TestLazyPessimisticUniqueCheck(t *testing.T) {
...
+	usage, err = telemetry.GetFeatureUsage(tk.Session())
+	require.NoError(t, err)
+	require.Equal(t, int64(0), usage.AddIndexLightning.AddIndexLightningUsed)
+
+	tk.MustExec("set @@global.tidb_ddl_enable_fast_reorg = on")
+	allow = ddl.IsEnableFastReorg()
+	require.Equal(t, true, allow)
+	usage, err = telemetry.GetFeatureUsage(tk.Session())
+	require.NoError(t, err)
+	// Because we have check once ddl.IsEnableFastReorg() so we get two as result.
```
Comment (tangenta) [inline telemetry/data_feature_usage_test.go:446]: The comment is inconsistent with the code.

File: metrics/telemetry.go [hunk 2]
```diff
@@ -284,3 +291,22 @@ func GetSavepointStmtCounter() int64 {
 func GetLazyPessimisticUniqueCheckSetCounter() int64 {
 	return readCounter(LazyPessimisticUniqueCheckSetCount)
 }
+
+// AddIndexLightning records the usages of Add Index with Lightning solution.
+type AddIndexLightningUsageCounter struct {
+	AddIndexLightningUsed int64 `json:"add_index_lightning_used"`
```
Comment (tangenta) [inline metrics/telemetry.go:297]: It is better to name it `add_index_ingest_used`

**Review by tangenta — APPROVED** [review]

**Review by AstroProfundis — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**